app/common/data/interfaces/collections.py

```diff
@@ -599,6 +622,32 @@ def as_flash_contexts(self) -> list[dict[str, str | bool]]:
         return flash_contexts
 
 
+class GroupContainsAddAnotherException(Exception, FlashableException):
+    def __init__(
+        self,
+        message: str,
+        group: Group,
+    ):
+        super().__init__(message)
+        self.message = message
+        self.group = group
+
+    def as_flash_context(self) -> dict[str, str | bool]:
+        contexts = self.as_flash_contexts()
+        return contexts[0] if contexts else {}
+
+    def as_flash_contexts(self) -> list[dict[str, str | bool]]:
+        flash_contexts = []
+        flash_context: dict[str, str | bool] = {
+            "message": self.message,
+            "group_name": self.group.name,
+            "group_id": str(self.group.id),
+            "grant_id": str(self.group.form.collection.grant_id),
```

Comment:
```suggestion
            "grant_id": str(self.form.collection.grant_id),
```

I think we can always go directly to the form?